parlai/agents/transformer/biencoder.py

```diff
@@ -8,7 +8,76 @@
 import torch
 
 
-class BiencoderAgent(TransformerRankerAgent):
+class AddLabelFixedCandsTRA(TorchRankerAgent):
```

Comment:
Typically call it a Trait, like `AddLabelFixedCandsTrait`, but in this case it looks like you could just have it be an abstract class and lose the multiple inheritance.

Reply:
I was imagining though that you'd have `Polyencoder(TorchRankerAgent)` and `IRFriendlyPolyencoder(AddLabelFixedCandsTRA, TorchRankerAgent)`